cocos/renderer/gfx-metal/MTLDevice.mm

```diff
@@ -287,4 +292,36 @@
     static_cast<CCMTLTexture*>(dst)->update(buffers.datas.data(), regions);
 }
 
+void CCMTLDevice::blitBuffer(void* srcData, uint offset, uint size, void* dstBuffer)
+{
+    id<MTLBuffer> sourceBuffer = id<MTLBuffer>(_blitedBuffer);
+    if(sourceBuffer == nil || sourceBuffer.allocatedSize < size)
+    {
+        if(sourceBuffer)
+            [sourceBuffer release];
+        sourceBuffer = [id<MTLDevice>(_mtlDevice) newBufferWithBytes:srcData
+                                                              length:size
+                                                             options:MTLResourceStorageModeShared];
+    }
+    
+    // Create a command buffer for GPU work.
+    id <MTLCommandBuffer> commandBuffer = [static_cast<View*>(_mtkView).mtlCommandQueue commandBuffer];
+    
+    
+    // Encode a blit pass to copy data from the source buffer to the private buffer.
+    id <MTLBlitCommandEncoder> blitCommandEncoder = [commandBuffer blitCommandEncoder];
+    [blitCommandEncoder copyFromBuffer:sourceBuffer
+                          sourceOffset:0
+                              toBuffer:id<MTLBuffer>(dstBuffer)
+                     destinationOffset:offset
+                                  size:size];
+    [blitCommandEncoder endEncoding];
+
+    // Add a completion handler and commit the command buffer.
+    [commandBuffer addCompletedHandler:^(id<MTLCommandBuffer> cb) {
+        // Private buffer is populated.
+    }];
```

Comment:
Do nothing in complete handler, why need it?

Reply:
to make sure source data did populated to GPU